go/vt/mysqlctl/backup.go

```diff
@@ -413,13 +413,14 @@ func backupFiles(mysqld MysqlDaemon, logger logutil.Logger, bh backupstorage.Bac
 	return nil
 }
 
-// checkNoDB makes sure there is no vt_ db already there. Used by Restore,
-// we do not wnat to destroy an existing DB.
+// checkNoDB makes sure there is no vt_ db already there.
+// If a dbNameOverride is specified, it checks for that instead.
+// Used by Restore, as we do not want to destroy an existing DB.
 // Returns true iff the condition is satisfied (there's no DB).
 // Returns (false, nil) if the check succeeds but the condition is not
 // satisfied (there is a DB).
 // Returns non-nil error if one occurs while trying to perform the check.
-func checkNoDB(ctx context.Context, mysqld MysqlDaemon) (bool, error) {
+func checkNoDB(ctx context.Context, mysqld MysqlDaemon, dbNameOverride string) (bool, error) {
```

Comment:
I think we should just pass down DbName, the real one. And at the higher level, use topoproto.TabletDbName.